### PR TITLE
Add Sendgrid

### PR DESCRIPTION
**Pull request by iNikitaGricenko**
Add sendgrid email notification plugin.
Need to figure out how to test that

### CODE REVIEW

**Comment by loicmathieu** [comment]
@iNikitaGricenko I rebased onto master so your branch only have commits related to Sendgrid

File: src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailSend.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+        this.to.stream().map(Email::new).forEach(personalization::addTo);
+
+        personalization.setSubject(runContext.render(this.subject));
+
+        Content plainTextContent = new Content(ContentType.TEXT_PLAIN.getMimeType(), "Please view this email in a modern email client");
+        mail.addContent(plainTextContent);
+
+        Content htmlContent = new Content(ContentType.TEXT_HTML.getMimeType(), runContext.render(this.htmlTextContent));
+        mail.addContent(htmlContent);
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailSend.java:154]: Should be rendered

File: src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailSend.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        }
+
+        if (this.embeddedImages != null) {
+            this.attachmentResources(this.embeddedImages, runContext).stream()
+                .peek(attachment -> attachment.setDisposition("inline"))
+                .forEach(mail::addAttachments);
+        }
+
+        if (this.cc != null) {
+            this.cc.stream().map(Email::new).forEach(personalization::addCc);
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailSend.java:169]: Should be rendered

**Review by loicmathieu — CHANGES_REQUESTED** [review]

File: src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailSend.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        title = "The optional subject of this email"
+    )
+    @PluginProperty(dynamic = true)
+    private String subject;
+
+    @Schema(
+        title = "The optional email message body in HTML text",
+        description = "Both text and HTML can be provided, which will be offered to the email client as alternative content" +
+            "Email clients that support it, will favor HTML over plain text and ignore the text body completely"
+    )
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailSend.java:140]: @iNikitaGricenko I think this is the last change that you didn't tackle, can you work on it so we can proceed with the PR?

**Comment by loicmathieu** [comment]
@iNikitaGricenko tests are failing can you have a look?

File: src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailSend.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (anna-geller) [reply src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailSend.java:140]: @loicmathieu can you add that missing text content? then we can merge this, as Nikita fixed the failing tests ✅

[the remark this answers is not in the document]
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailSend.java]: Done

**Review by loicmathieu — APPROVED** [review]
LGTM, I made some small adaptation

**Pull request merged**